File: Daily/MEXAfterOperations.py
from typing import List
from collections import defaultdict
# 2598
class Solution:
    def findSmallestInteger(self, nums: List[int], value: int) -> int:

        freq = defaultdict(int)
        for i in range(len(nums)):
            rem = nums[i] % value
            freq[rem] += 1
        
        i = 0
        while True:
            if freq[i % value] == 0:
                return i
            freq[i % value] -= 1
            i += 1

File: Java/Daily/Medium/MEXAfterOperations.java
import java.util.*;
// 2598
class Solution {
    public int findSmallestInteger(int[] nums, int value) {
        
        Map<Integer, Integer> freq = new HashMap<>();
        for ( int i = 0; i < nums.length; i++) {
            int rem = ((nums[i] % value) + value) % value;
            freq.put(rem, freq.getOrDefault(rem, 0) + 1);
        }
        int i = 0;
        while (true) {
            int rem = i % value;
            int count = freq.getOrDefault(rem, 0);
            if (count == 0)
                return i;
            freq.put(rem, count - 1);
            i++;
        }
    }
}